cli/src/composer.rs

```diff
@@ -0,0 +1,211 @@
+use std::collections::HashMap;
+
+use failure::*;
+use rusqlite::Connection;
+use serde_json::Value;
+use trustnote::parent_composer::LastStableBallAndParentUnits;
+use trustnote::spec::{Message, Payload, Payment};
+use trustnote::*;
+
+struct Param {
+    signing_addresses: Vec<String>,
+    paying_addresses: Vec<String>,
+    outputs: Vec<spec::Output>,
+    messages: Vec<spec::Message>,
+    signer: String,
+    light_props: Option<LastStableBallAndParentUnits>,
+    unit: spec::Unit,
+}
+
+// TODO: params name
+#[allow(dead_code)]
+fn compose_joint(mut params: Param) -> Result<()> {
+    let witnesses = params.unit.witnesses.clone();
+    if witnesses.is_empty() {
+        params.unit.witnesses = my_witness::read_my_witnesses()?;
+        return compose_joint(params);
+    }
+
+    if params.light_props.is_none() {
+        match request_from_light_vendor(
+            "light/get_parents_and_last_ball_and_witness_list_unit",
```

Comment:
this should call a specific network function and not use the string message as param